### PR TITLE
HOTFIX - Refactor offline users change handling (deleted removeBy)

### PR DESCRIPTION
**Pull request by ruimfernandes**
### Refactor offline users change handling (deleted removeBy)
I don't know why but `.splice()` function wasn't doing the side effect on the array (removing the element) and was only returning the removed element which was pretty useless.

In order to make necessary changes at `videoCallClient.users` I replaced the logic for a working one 😉 

[https://www.w3schools.com/jsref/jsref_splice.asp](https://www.w3schools.com/jsref/jsref_splice.asp)
If you check Definition and Usage it says that original array is modified

**PS: IGNORE `dist/browser.js`... It's bower build related stuff**
### :point_right: Scroll until the end and check :point_left:
- `lib/handler.js`
- `lib/helpers/remove-by.js`
- `test/unit/helpers/remove-by-test.js`

### CODE REVIEW

**Comment by zpr-sioslife** [comment]
splice() returns the deleted element

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

**Comment by zpr-sioslife** [comment]
`remove-by.js` was hot garbage. Glad it's gone.

File: lib/handler.js [hunk 1]
```diff
@@ -56,8 +55,14 @@ function Handler(videoCallClient, clientType) {
...
 
   function _handleLeavingUser(message) {
-    var removed = removeBy(videoCallClient.users, message.user, 'userId');
-    if(removed) {
+    var initialLength = videoCallClient.users.length;
+    videoCallClient.users = videoCallClient.users.filter(function(user) {
+      if(user.userId !== message.user.userId) {
+        return user;
+      }
+    });
```
Comment (zpr-sioslife) [inline lib/handler.js]:
```js
videoCallClient.users = videoCallClient.users.filter(function(user) {
  return user.userId !== message.user.userId
});
```

Cleaner. filter() creates a new array with all elements that pass the test implemented by the provided function (True/False)

Comment (ruimfernandes) [reply lib/handler.js]: Yup, nice catch 😉 
I've started using `.map()` and that's why I left the `return user` 🤦‍♂️

**Pull request merged**